cmd/podman/replay_kube.go

```diff
@@ -0,0 +1,247 @@
+			Usage: "`pathname` of a directory containing TLS certificates and keys",
+		},
+		cli.StringFlag{
+			Name:  "creds",
+			Usage: "`credentials` (USERNAME:PASSWORD) to use for authenticating to a registry",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "Suppress output information when pulling images",
+		},
+		cli.StringFlag{
+			Name:  "signature-policy",
+			Usage: "`pathname` of signature policy file (not usually used)",
+		},
+		cli.BoolTFlag{
+			Name:  "tls-verify",
+			Usage: "require HTTPS and verify certificates when contacting registries (default: true)",
+		},
+	}
+	replayKubeDescription = "Replay a Pod and its containers based on a Kubrernetes YAML"
+	replayKubeCommand     = cli.Command{
+		Name:                   "kube",
+		Usage:                  "Replay a pod based on Kubenetes YAML",
+		Description:            replayKubeDescription,
+		Action:                 replayKubeYAMLCmd,
+		Flags:                  sortFlags(replayKubeFlags),
+		ArgsUsage:              "kubernetes YAML file",
+		UseShortOptionHandling: true,
+		OnUsageError:           usageErrorHandler,
+	}
+)
+
+// generateKubeYAMLCmdgenerates or replays kube
+func replayKubeYAMLCmd(c *cli.Context) error {
+	var (
+		podOptions    []libpod.PodCreateOption
+		podYAML       v1.Pod
+		writer        io.Writer
+		registryCreds *types.DockerAuthConfig
+		namespaces    map[string]string
+		containers    []*libpod.Container
+	)
+
+	ctx := getContext()
+	if rootless.IsRootless() {
+		return errors.Wrapf(libpod.ErrNotImplemented, "rootless users")
+	}
+	args := c.Args()
+	if len(args) > 1 {
+		return errors.New("you can only replay one kubernetes file at a time")
+	}
+	if len(args) < 1 {
+		return errors.New("you must supply at least one file")
+	}
+
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	content, err := ioutil.ReadFile(args[0])
+	if err != nil {
+		return err
+	}
+
+	if err := yaml.Unmarshal(content, &podYAML); err != nil {
+		return errors.Wrapf(err, "unable to read %s as YAML", args[0])
+	}
+
+	podOptions = append(podOptions, libpod.WithInfraContainer())
+	podOptions = append(podOptions, libpod.WithPodName(podYAML.ObjectMeta.Name))
+	// TODO for now we just used the default kernel namespaces; we need to add/subtract this from yaml
+
+	nsOptions, err := shared.GetNamespaceOptions(strings.Split(DefaultKernelNamespaces, ","))
+	if err != nil {
+		return err
+	}
+	podOptions = append(podOptions, nsOptions...)
+	podPorts := getPodPorts(podYAML.Spec.Containers)
+	podOptions = append(podOptions, libpod.WithInfraContainerPorts(podPorts))
+
+	// Create the Pod
+	pod, err := runtime.NewPod(ctx, podOptions...)
+	if err != nil {
+		return err
+	}
+	// Print the Pod's ID
+	fmt.Println(pod.ID())
+
+	podInfraID, err := pod.InfraContainerID()
+	if err != nil {
+		return err
+	}
+
+	namespaces = map[string]string{
+		"pid":  fmt.Sprintf("container:%s", podInfraID),
```

Comment:
We only want net, ipc, uts - not user or pid

Reply:
Well... We might want userns, actually. But for now, let's pretend userns doesn't exist, it's much easier that way

Reply:
for my own knowledge, why ?

Reply:
the regular pod create deals with those, hence my question

Reply:
PID namespace, we don't share in pods because Kube doesn't share it by default. User namespace, we pretend it doesn't exist because it's a real pain to deal with - we have them working well, but not for dependency scenarios.

Reply:
OK, commenting out and noting as such